apps/src/applab/applab.js

```diff
@@ -689,15 +690,31 @@ Applab.init = function(config) {
 
   if (config.level.makerlabEnabled) {
     makerToolkit.enable();
+
     config.dropletConfig = utils.deepMergeConcatArrays(
       config.dropletConfig,
       makerToolkit.dropletConfig
     );
+
+    if (config.level.makerlabEnabled === MB_API) {
+      config.dropletConfig = utils.deepMergeConcatArrays(
+        config.dropletConfig,
+        makerToolkit.configMicrobit
+      );
+    } else {
+      config.dropletConfig = utils.deepMergeConcatArrays(
+        config.dropletConfig,
+        makerToolkit.configCircuitPlayground
+      );
+    }
   } else {
-    // Push gray, no-autocomplete versions of maker blocks for display purposes.
-    const disabledMakerDropletConfig = makeDisabledConfig(
-      makerToolkit.dropletConfig
+    const makerConfig = utils.deepMergeConcatArrays(
+      makerToolkit.dropletConfig,
+      makerToolkit.configCircuitPlayground
```

Comment:
i might be misunderstanding this conditional, but do we want `makerToolkit.configCircuitPlayground` here?

i see:
```javascript
if (config.level.makerlabEnabled) {
  ...
} else {
  const makerConfig = utils.deepMergeConcatArrays(
      makerToolkit.dropletConfig,
      makerToolkit.configCircuitPlayground
  );
}
```

if `config.level.makerlabEnabled` is false, i wouldn't expect that we would be merging the circuit playground config here

Reply:
better question: what does the `config.level.makerlabEnabled` mean? 😄

Reply:
The 'config.level.makerlabEnabled' holds the string of which maker API is enabled, so 'circuitPlayground', 'microBit', or 'null'. There is a bit of a labyrinth of how that value gets sets, that I'm happy to walk through if you are interested, but didn't want to bog this comment down with it unless needed.

So here, when the there is an API enabled and config.level.makerlabEnabled is either 'microbit' or 'circuitPlayground', we go ahead and use one of those droplet configs. When that variable is null, we drop into the else where we gather all the maker droplet configs as seen in line 711. That maker config is then disabled on line 717, which includes removing the blocks and categories. Then on line 718, the droplet config is reset to a new value with the maker config disabled. So when the makerlabEnabled is false, we want to use the circuit playground config to maker sure we are disabling/removing those blocks from the droplet config.

Reply:
Follow-up question for @breville - is there a reason why in the 'enabling' part of this if/else we use either configMicrobit or configCircuitPlayground, but in the else branch, we just merge the configCircuitPlayground? I would expect that would lead us to issues disabling the microBit blocks, but that doesn't seem the case . . .

Reply:
that makes sense to me -- thank you for explaining! there was also a very important piece of info that i missed before -- all of this logic is wrapped in an `if (config.level.makerlabEnabled) {}` block 🙃

Reply:
@epeach I'll defer to your expertise on what we should do in this code path.  Happy to pair up and test all scenarios.  I think I was trying to maintain existing behaviour, but do we have to account for cases in which an app is authored for one piece of hardware and then reopened later but there is no hardware present, or a different type of hardware is being used?